### DISCUSSION TITLE
How to open port on host machines for a new load balancer

### DISCUSSION THREAD

**Discussion by Biacode** [mysticaltech/terraform-hcloud-kube-hetzner#416] [Q&A] [answered]
Hello,
How can I expose a new load-balancer using `type: LoadBalancer`?
It creates a new one but never gets to a healthy state. See the screenshot above.
I assume the issue is that all ports are blocked on agents except the essential ones that use k8s?
To give you more context, I am trying to [deploy the following service.](https://github.com/l7mp/stunner)
![image](https://user-images.githubusercontent.com/10510996/201793723-aabc6d13-5fda-4b41-b10a-ff0c312b3f54.png)

**Comment by Biacode** [comment]
Update: Just for testing purposes, I removed all firewall rules and removed the firewall itself to make that at least the firewall doesn't block the ports. But still, no luck.

**Comment by Biacode** [comment]
@mysticaltech more screenshots if those could help
![image](https://user-images.githubusercontent.com/10510996/201907413-fe1e6fcd-b575-4acd-bafd-c2fd466e3a88.png)

**Comment by Biacode** [comment]
UPDATE: I managed to have a "healthy" load balancer. See the image below.
The issue is that the Hetzner LB doesn't support UDP, so instead, I ended up having only TCP.
Does anyone know how I can change this behavior?
E.g., instead of using Hetzner LB, use a different LB that supports UDP
I believe there is Klipper LB in this distribution. But I am not sure if that will work with UDP. Also not sure if it will work with my cert-manager and HTTPS services.
![image](https://user-images.githubusercontent.com/10510996/202192610-df47c6e8-55b3-4dcf-bda1-01b940672f4f.png)

**Reply by Biacode** [reply]
> Hey @Biacode, interesting. However, here's what I recommend:
> 
>     * Never touch the Hetzner UI, instead get to know the `hcloud` command, just to read info in your case like `hcloud load-balancer list` and `hcloud load-balancer describe <name>`. As to not risk messing around with either the Kubernetes or the Terraform state.
> 
>     * If you want UDP, you have to use Traefik, not Nginx, the good thing is that it is the default, and proxy protocol (also turned on by default) ensures that you do not have to configure the LB at all (Just DO NOT TOUCH IT), instead, you want to configure TRAEFIK as it is the ingress controller, and to put it in simple terms, IT IS THE BIG BOSS OF THE LB 😁, it will configure it the way it needs! So get to know the traefik docs, like this one https://doc.traefik.io/traefik/routing/entrypoints/.
> 
>     * For UDP, I believe you need to add another "entrypoint" and that's all, please search our project for examples of how to create a HelmChartConfig to reconfigure Traefik. You can also do it via the `traefik_values` variable on first run, but best to get to know how to reconfigure it while it's running. Basically, just pull down the helmchartconfig with `kubectl get helmchartconfig <traefik or similar> -o yaml`, edit it, and apply it. Note that you can also use kubectl edit, but not very GitOps friendly.
> 
>     * Then the final point you can maybe configure is your ingress definition, get to know that, especially for Traefik. Again lots of examples if you search this repo.

Thanks for the guidance, @mysticaltech!

**Comment by mysticaltech** [comment, marked as answer]
To recap for future readers:

- Never touch the Hetzner UI, instead get to know the hcloud command, just to read info in your case like hcloud load-balancer list and hcloud load-balancer describe <name>. As to not risk messing around with either the Kubernetes or the Terraform state.

- If you want UDP, you have to use Traefik, not Nginx, the good thing is that it is the default, and proxy protocol (also turned on by default) ensures that you do not have to configure the LB at all (Just DO NOT TOUCH IT), instead, you want to configure TRAEFIK as it is the ingress controller, and to put it in simple terms, IT IS THE BIG BOSS OF THE LB grin, it will configure it the way it needs! So get to know the traefik docs, like this one https://doc.traefik.io/traefik/routing/entrypoints/.

- For UDP, I believe you need to add another "entrypoint" and that's all, please search our project for examples of how to create a HelmChartConfig to reconfigure Traefik. You can also do it via the traefik_values variable on first run, but best to get to know how to reconfigure it while it's running. Basically, just pull down the helmchartconfig with kubectl get helmchartconfig <traefik or similar> -o yaml, edit it, and apply it. Note that you can also use kubectl edit, but not very GitOps friendly.

- Then the final point you can maybe configure is your ingress definition, get to know that, especially for Traefik. Again lots of examples if you search this repo.
